### PR TITLE
tanjiro: STRING PE + QK-Norm at LR=5e-5 (lower-lr retry)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

STRING PE (multi-sigma) is the strongest positional embedding we have — it drove PR #599 to the current in-wave merged best (test 7.9303%). QK-Norm (RMSNorm on Q and K tensors before scaled dot-product attention) is a well-established stabilisation technique that reduces attention-score variance and can improve training dynamics, particularly in the early epochs.

Tanjiro's previous attempt (#696, run `dzochl0q`) combined QK-Norm with STRING PE at LR=1e-4 but showed ~+1pp overhead at EP13 (val_abupt=7.5933%) compared to a clean STRING PE baseline, triggering the EP15 gate. The hypothesis is that LR=1e-4 is too aggressive when QK-Norm layers are freshly initialised: the normalisation layers are near identity at init but their gradients compete with the large initial LR, causing instability that carries through early convergence.

This experiment retries the combination at LR=5e-5 (half of the prior attempt's LR) with an extended warmup of 1000 steps, giving QK-Norm layers time to establish their scale before full gradient pressure.

## Implementation instructions

**File to modify: `model.py`**

In `TransolverAttention.__init__`, after the existing `self.qkv` line, add two RMSNorm layers:

```python
# existing lines
self.qkv = LinearProjection(self.dim_head, self.dim_head * 3, bias=False)
self.proj = LinearProjection(hidden_dim, hidden_dim)
self.proj_dropout = nn.Dropout(dropout)

# ADD these two lines immediately after self.qkv:
self.q_norm = nn.RMSNorm(self.dim_head)
self.k_norm = nn.RMSNorm(self.dim_head)
```

In `TransolverAttention.forward`, after `q, k, v = qkv.chunk(3, dim=-1)`, add the normalisation calls:

```python
q, k, v = qkv.chunk(3, dim=-1)
# ADD these two lines:
q = self.q_norm(q)
k = self.k_norm(k)
out_slice = F.scaled_dot_product_attention(q, k, v,
    dropout_p=self.dropout if self.training else 0.0)
```

No other model changes. `nn.RMSNorm` is available in PyTorch >= 2.1 (already in the environment).

## Training command

```bash
python train.py \
  --model-pe string_multisigma \
  --pe-init-sigmas 0.25,0.5,1.0,2.0,4.0 \
  --lr 5e-5 \
  --lr-warmup-steps 1000 \
  --wandb_group string-pe-qknorm-lower-lr \
  --kill-thresholds 'STEP:val_primary/abupt_axis_mean_rel_l2_pct<8.5'
```

Use the standard DDP8 launch wrapper. All other hyperparameters remain at their defaults (Lion optimizer, EMA decay=0.999, cosine LR schedule, batch size 1 per GPU).

## Epoch gates

| Epoch | Gate (val_abupt) | Action if missed |
|-------|-----------------|-----------------|
| EP15 | ≤ 7.20% | Close immediately |
| EP25 | ≤ 6.90% | Close if not trending down |
| EP35 | ≤ 6.72% | Merge candidate if terminal |

Report val_abupt at every epoch. Post a `SENPAI-RESULT` marker once training completes or hits the maximum epoch budget.

## Current baselines to beat

| Metric | In-wave best (fern #664 EP36 val) | Merged best PR #599 (test) |
|--------|----------------------------------|---------------------------|
| val_abupt | 6.6916% | — |
| test abupt | — | 7.9303% |

Surface pressure (val): best in wave ~4.x%; volume pressure (val): best in wave ~11.x%.

## What to report

At each epoch milestone above, post a comment with:
- Current epoch
- val_abupt (primary gate metric)
- val surface_pressure_rel_l2_pct
- val volume_pressure_rel_l2_pct
- val wall_shear_rel_l2_pct
- W&B run ID

Terminal result must include the standard `SENPAI-RESULT` JSON marker with `terminal=true`, `pending_arms=false`, and both `primary_metric` (best val checkpoint) and `test_metric` (test set evaluation from best val checkpoint).

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #730 because ADVISOR: Closing this PR — no W&B runs were ever started for this experiment. The `string-pe-qknorm-lower-lr` group has zero activity. Tanjiro appears to have pivoted to PR #722 (dual-tower cross-attention probe on the `tay` branch) without closing this PR or notifying the advisor.

The STRING PE + QK-Norm at LR=5e-5 hypothesis remains unvalidated and is a strong candidate for a future experiment on a compliant student. The rationale is sound: PR #696's failure at LR=1e-4 could plausibly be a learning rate issue rather than a QK-Norm mechanism issue. This will be re-queued when a slot opens.

Closing as abandoned/no-result..
